x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/search_panel/search_panel.tsx

```diff
@@ -61,98 +63,112 @@ export const SearchPanel: FC<Props> = ({
+      queryManager.filterManager.removeAll();
+    };
+  }, [searchQueryLanguage, searchString, queryManager.filterManager]);
+
+  const searchHandler = ({ query, filters }: { query?: Query; filters?: Filter[] }) => {
+    const mergedQuery = query ?? searchInput;
     try {
-      if (query.language === SEARCH_QUERY_LANGUAGE.KUERY) {
-        filterQuery = toElasticsearchQuery(fromKueryExpression(query.query), indexPattern);
-      } else if (query.language === SEARCH_QUERY_LANGUAGE.LUCENE) {
-        filterQuery = luceneStringToDsl(query.query);
-      } else {
-        filterQuery = {};
+      if (filters) {
+        queryManager.filterManager.setFilters(filters);
       }
+
+      const combinedQuery = createMergedEsQuery(
+        mergedQuery,
+        queryManager.filterManager.getFilters() ?? [],
+        indexPattern,
+        uiSettings
+      );
+
       setSearchParams({
-        searchQuery: filterQuery,
-        searchString: query.query,
-        queryLanguage: query.language as SearchQueryLanguage,
+        searchQuery: combinedQuery,
+        searchString: mergedQuery.query,
+        queryLanguage: mergedQuery.language as SearchQueryLanguage,
       });
     } catch (e) {
       console.log('Invalid syntax', JSON.stringify(e, null, 2)); // eslint-disable-line no-console
-      setErrorMessage({ query: query.query as string, message: e.message });
+      toasts.addError(e, {
+        title: i18n.translate('xpack.dataVisualizer.searchPanel.invalidSyntax', {
+          defaultMessage: 'Invalid syntax',
+        }),
+      });
     }
   };
-  const searchChangeHandler = (query: Query) => setSearchInput(query);
 
   return (
-    <EuiFlexGroup gutterSize="m" alignItems="center" data-test-subj="dataVisualizerSearchPanel">
-      <EuiFlexItem>
-        <EuiInputPopover
-          style={{ maxWidth: '100%' }}
-          closePopover={() => setErrorMessage(undefined)}
-          input={
-            <QueryStringInput
-              bubbleSubmitEvent={false}
-              query={searchInput}
-              indexPatterns={[indexPattern]}
-              onChange={searchChangeHandler}
-              onSubmit={searchHandler}
-              placeholder={i18n.translate(
-                'xpack.dataVisualizer.searchPanel.queryBarPlaceholderText',
-                {
-                  defaultMessage: 'Search… (e.g. status:200 AND extension:"PHP")',
-                }
-              )}
-              disableAutoFocus={true}
-              dataTestSubj="dataVisualizerQueryInput"
-              languageSwitcherPopoverAnchorPosition="rightDown"
-            />
+    <EuiFlexGroup
+      gutterSize="s"
+      alignItems="flexStart"
+      data-test-subj="dataVisualizerSearchPanel"
+      className={'dvSearchPanel'}
+      responsive={false}
+    >
+      <EuiFlexItem grow={9} className={'dvSearchBar'}>
+        <SearchBar
+          dataTestSubj="dataVisualizerQueryInput"
+          appName={'dataVisualizer'}
+          showFilterBar={true}
```

Comment:
When opening the link to Discover, the search part of the search bar is retained, but the filter is lost. Is it possible to pass the filter part too? Same applies for the link to Lens.

Reply:
Fixed here https://github.com/elastic/kibana/pull/112870/commits/6c0c8505076c4152d0ddacfb34b3909598c08403. It should now also update the filters for Lens and Maps as well.

Reply:
If I create a saved search containing a filter in Discover, then open that search in the Data Visualizer and drilldown to Lens, I get this error in Lens:

![image](https://user-images.githubusercontent.com/7405507/135294432-0756cdab-ccbd-4df6-8c6a-ec62f7d47eaf.png)

(Works fine if I add the filter inside the Data Visualizer and then drilldown to Lens).